source/common/buffer/buffer_impl.cc

```diff
@@ -124,44 +123,31 @@ void OwnedImpl::move(Instance& rhs, uint64_t length) {
   static_cast<LibEventInstance&>(rhs).postProcess();
 }
 
-Api::SysCallIntResult OwnedImpl::read(int fd, uint64_t max_length) {
+Network::IoHandleCallUintResult OwnedImpl::read(Network::IoHandle& io_handle, uint64_t max_length) {
   if (max_length == 0) {
-    return {0, 0};
+    return Network::IoHandleCallUintResult(0,
+                                           Network::IoErrorPtr(nullptr, [](Network::IoError*) {}));
```

Comment:
This is just "no error", right? Can we add a "no error" constructor variant to IoHandleCallUintResult or maybe some other short cut to hide this at every call site?

Reply:
IoHandleCallResult is a template class. It's not straight forward to add a constructor to initialize rc_ to be 0. Instead I defined a `IO_CALL_RESULT_NO_ERROR` macro there.